server/docker/Dockerfile

```diff
@@ -1,5 +1,84 @@
+#
+# Setup Builder Image
+#
+########################################################################################################################
+FROM ubuntu:${UBUNTU_TAG} AS openjdk-builder
+
+# Define Standard Environment Variables
+ENV LC_ALL=C.UTF-8
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Install basic OS utilities for building
+RUN apt-get update && \
+	apt-get install --yes tar gzip gnupg2 curl
+
+##########################
+####    Java Setup    ####
+##########################
+
+RUN set -eux; \
+        ARCH="$(dpkg --print-architecture)"; \
+        case "${ARCH}" in \
+           aarch64|arm64) \
+             ESUM='e184dc29a6712c1f78754ab36fb48866583665fa345324f1a79e569c064f95e9'; \
+             BINARY_URL='https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.1%2B12/OpenJDK21U-jdk_aarch64_linux_hotspot_21.0.1_12.tar.gz'; \
+             ;; \
+           amd64|i386:x86-64) \
+             ESUM='1a6fa8abda4c5caed915cfbeeb176e7fbd12eb6b222f26e290ee45808b529aa1'; \
+             BINARY_URL='https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.1%2B12/OpenJDK21U-jdk_x64_linux_hotspot_21.0.1_12.tar.gz'; \
+             ;; \
+           ppc64el|powerpc:common64) \
+             ESUM='9574828ef3d735a25404ced82e09bf20e1614f7d6403956002de9cfbfcb8638f'; \
+             BINARY_URL='https://github.com/adoptium/temurin21-binaries/releases/download/jdk-21.0.1%2B12/OpenJDK21U-jdk_ppc64le_linux_hotspot_21.0.1_12.tar.gz'; \
+             ;; \
+           *) \
+             echo "Unsupported arch: ${ARCH}"; \
+             exit 1; \
+             ;; \
+        esac; \
+    curl -LfsSo /tmp/openjdk.tar.gz ${BINARY_URL}; \
+    echo "${ESUM} */tmp/openjdk.tar.gz" | sha256sum -c -; \
+    mkdir -p /usr/local/java; \
+    tar --extract \
+    	      --file /tmp/openjdk.tar.gz \
+    	      --directory "/usr/local/java" \
+    	      --strip-components 1 \
+    	      --no-same-owner \
+    	  ; \
+    rm -f /tmp/openjdk.tar.gz /usr/local/java/lib/src.zip;
+
+########################################################################################################################
+#
+# Build Final Image
+#
+########################################################################################################################
+FROM ubuntu:${UBUNTU_TAG} AS openjdk-base
+
+# Define Standard Environment Variables
+ENV LC_ALL=C.UTF-8
+ENV DEBIAN_FRONTEND=noninteractive
+ENV JAVA_VERSION="jdk-21.0.1+12"
+ENV JAVA_HOME="/usr/local/java/"
+
+# Fetch Validated Java Binaries
+COPY --from=openjdk-builder /usr/local/java/ /usr/local/java/
+
+# Install Basic OS Requirements
+RUN apt-get update && \
+	apt-get install --yes --no-install-recommends tar gzip openssl curl && \
+	apt-get autoremove --yes && \
+	apt-get autoclean --yes && \
+	apt-get clean all --yes && \
+    rm -rf /var/lib/{apt,dpkg,cache,log}/
+
```

Comment:
We should probably work with release engineering to have a baseline image (with the JDK and tool installs added) published to our registry on a regular basis instead of doing all these updates in every single build.
It would remove a surprising amount of time and cost from the thousands of these builds (across multiple projects) that run each month...